services/actions/src/rpc/checkAlert.js

```diff
@@ -0,0 +1,156 @@
+import { fetchData } from './fetchDataset';
+import sendExplorationScreenshot from './sendExplorationScreenshot';
+
+import apiError from '../utils/apiError';
+import generateUserAccessToken from '../utils/jwt';
+import { fetchGraphQL } from '../utils/graphql';
+import redisClient from '../utils/redis';
+
+const alertQuery = `
+  query ($id: uuid!) {
+    alerts_by_pk(id: $id) {
+      id
+      name
+      trigger_config
+      delivery_type
+      delivery_config
+      exploration {
+        id
+        datasource_id
+        user_id
+        playground_state
+      }
+    }
+  }
+`;
+
+const checkAndTriggerAlert = async (alert) => {
+  const result = {
+    fired: false,
+    locked: false,
+    lockKey: null,
+    lockValue: null,
+    lockTTL: 0,
+  };
+
+  const {
+    id,
+    name,
+    exploration,
+    trigger_config: triggerConfig,
+    delivery_type: deliveryType,
+    delivery_config: deliveryConfig,
+  } = alert;
+  const { playground_state: playgroundState, user_id: userId } = exploration;
+
+  const lockKey = `alert:${id}:lock`;
+  const lockValue = await redisClient.get(lockKey);
```

Comment:
it will not work for the `mlcraft-stack`, should work somehow if redis is not provided